app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -711,10 +711,12 @@ def build_toolbar_hide_button(id)
       return false if MIQ_AE_COPY_ACTIONS.include?(id) && User.current_tenant.any_editable_domains? && MiqAeDomain.any_unlocked?
       case id
       when "miq_ae_domain_lock"
-        return true unless editable_domain?(@record)
+        return true unless @record.editable_properties? && !@record.contents_locked?
```

Comment:
@Fryguy The context of this is the UI, where they want to show "Unlock" menu option only if the domain is locked and show the "Lock" menu option if it is unlocked.